src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -834,6 +818,20 @@ private bool InferConnectionDetails()
                     ReportSNIError(SNIProviders.INVALID_PROV);
                     return false;
                 }
+
+                try
```

Comment:
Another recommendation I can make is, create a Property called `IsSSRPrequired` in `DataSource` and set it when you can conclude from Parsing logic that SSRP is needed. 
Then use the property in the `CreateConnectionHandle ` to decide if you need to query the browser.